deps/http_parser.mk

```diff
@@ -0,0 +1,39 @@
+## http_parser
+
+HTTPPARSER_GIT_URL := git://github.com/nodejs/http-parser.git
+HTTPPARSER_TAR_URL = https://api.github.com/repos/nodejs/http-parser/tarball/$1
+$(eval $(call git-external,http_parser,HTTPPARSER,,,$(BUILDDIR)))
+
+# Manually set the PLATFORM flag to whatever $(OS) is, because at least one
+# person (@staticfloat) has the PLATFORM envvar set to something else.
+# Also manually set PREFIX to DESTDIR
+HTTPPARSER_FLAGS := PLATFORM=$(shell echo $(OS) | tr A-Z a-z) PREFIX=$(build_prefix)
+
+$(eval $(call staged-install, \
+	http_parser,$$(HTTPPARSER_SRC_DIR), \
+	MAKE_INSTALL,$(HTTPPARSER_FLAGS),, \
+	$(INSTALL_NAME_CMD)libhttp_parser.$(SHLIB_EXT) $(build_shlibdir)/libhttp_parser.$(SHLIB_EXT)))
+
+$(BUILDDIR)/$(HTTPPARSER_SRC_DIR)/http_parser-makefile.patch-applied: $(BUILDDIR)/$(HTTPPARSER_SRC_DIR)/source-extracted
+	cd $(BUILDDIR)/$(HTTPPARSER_SRC_DIR) && patch -p1 -f < $(SRCDIR)/patches/http_parser-makefile.patch
```

Comment:
Upstream has had something similar to this patch open as a [PR since June](https://github.com/nodejs/http-parser/pull/321), it looks like upstream is a little slow to respond to PRs.  The patch carried here is based off of the patch in that PR, with some extra stuff for Julia.  I'll see if I can pare the patch down, I was just copying it wholesale from wildart's original work.